modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -162,42 +504,797 @@ public void testListWithGetInt(String filter, int startIndexInt, int countInt,
 
     @Test(dataProvider = "dataForListWithGetInteger")
     public void testListWithGetInteger(String filter, Integer startIndexInt,
-                        Integer countInt, String sortBy, String sortOrder, String domainName,
-                        String attributes, String excludeAttributes, int expectedScimResponseStatus) {
-
+                                       Integer countInt, String sortBy, String sortOrder, String domainName,
+                                       String attributes, String excludeAttributes, int expectedScimResponseStatus) {
 
         SCIMResponse outputScimResponse = userResourceManager.listWithGET(userManager, filter, startIndexInt,
                 countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
         Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
     }
 
-    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+    @DataProvider(name = "dataForTestCreateUserSuccess")
+    public Object[][] dataToTestCreateUserSuccess() {
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null},
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", "emails"}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateUserSuccess")
+    public void testCreateUserSuccess(String scimObjectString, String attributes, String excludeAttributes)
+            throws BadRequestException, CharonException, InternalErrorException, ConflictException {
+
+        User user = getNewUser();
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        Mockito.when(userManager.createUser(any(User.class), anyMap())).thenReturn(user);
+
+        SCIMResponse scimResponse = userResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(scimResponse.getResponseMessage());
+
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_CREATED);
```

Comment:
I have noticed that in some methods the response status has been passed from the data provider and in others, it is directly called inside the method. It would be nice to have it used in a consistent manner.

Reply:
That was previous author, Changed in commit 6f5ca461fa07ac4b49182320325156a58b80b910